Disable Resteasy default method security authorization

When a Flow view is annotated with a security annotation (e.g. `@PermitAll`), Resteasy performs an authorization check on every public method, and this sometimes fails because of a missing security RequestScoped bean.

```
2023-04-22 17:02:23,212 ERROR [rea.cor.pub.Operators] (parallel-1) Operator called default onErrorDropped: reactor.core.Exceptions$ErrorCallbackNotImplemented: javax.enterprise.context.ContextNotActiveException: RequestScoped context was not active when trying to obtain a bean instance for a client proxy of CLASS bean [class=io.quarkus.vertx.http.runtime.CurrentVertxRequest, id=9260aa608251c44230eeb9b234f605844a7622bf]
	- you can activate the request context for a specific method using the @ActivateRequestContext interceptor binding
Caused by: javax.enterprise.context.ContextNotActiveException: RequestScoped context was not active when trying to obtain a bean instance for a client proxy of CLASS bean [class=io.quarkus.vertx.http.runtime.CurrentVertxRequest, id=9260aa608251c44230eeb9b234f605844a7622bf]
	- you can activate the request context for a specific method using the @ActivateRequestContext interceptor binding
	at io.quarkus.arc.impl.ClientProxies.getDelegate(ClientProxies.java:55)
	at io.quarkus.vertx.http.runtime.CurrentVertxRequest_ClientProxy.arc$delegate(Unknown Source)
	at io.quarkus.vertx.http.runtime.CurrentVertxRequest_ClientProxy.getOtherHttpContextObject(Unknown Source)
	at io.quarkus.resteasy.reactive.server.runtime.QuarkusCurrentRequest.get(QuarkusCurrentRequest.java:21)
	at org.jboss.resteasy.reactive.server.core.CurrentRequestManager.get(CurrentRequestManager.java:8)
	at io.quarkus.resteasy.reactive.server.runtime.StandardSecurityCheckInterceptor.intercept(StandardSecurityCheckInterceptor.java:38)
	at io.quarkus.resteasy.reactive.server.runtime.StandardSecurityCheckInterceptor_RolesAllowedInterceptor_Bean.intercept(Unknown Source)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:42)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:33)
	at com.example.application.flow.FlowView_Subclass.getUI(Unknown Source)
	at com.example.application.flow.FlowView.updateDiv(FlowView.java:46)
	at reactor.core.publisher.LambdaSubscriber.onNext(LambdaSubscriber.java:160)
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onNext(FluxOnErrorResume.java:79)
	at reactor.core.publisher.FluxPeek$PeekSubscriber.onNext(FluxPeek.java:200)
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:120)
	at reactor.core.publisher.FluxOnBackpressureDrop$DropSubscriber.onNext(FluxOnBackpressureDrop.java:128)
	at reactor.core.publisher.FluxInterval$IntervalRunnable.run(FluxInterval.java:125)
	at reactor.core.scheduler.PeriodicWorkerTask.call(PeriodicWorkerTask.java:59)
	at reactor.core.scheduler.PeriodicWorkerTask.run(PeriodicWorkerTask.java:73)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

This check is not required for Flow or Hilla, as they have their own checks.
To prevent the issue, register a custom `AuthorizationController` in the extension, sot that authorization can be enabled or disabled in the configuration file.
By default, authorization will be disabled.


```
@Alternative
@Priority(Interceptor.Priority.LIBRARY_AFTER)
@ApplicationScoped
public class DisabledAuthController extends AuthorizationController {
    @ConfigProperty(name = "hilla.security.disable-quarkus-authorization", defaultValue = "true")
    boolean disableAuthorization;

    @Override
    public boolean isAuthorizationEnabled() {
        return !disableAuthorization;
    }
}
```

--- a/code-with-quarkus/src/main/java/com/example/application/ClockService.java
+++ b/code-with-quarkus/src/main/java/com/example/application/ClockService.java
@@ -7,15 +7,18 @@ import java.util.UUID;
 
 import dev.hilla.Nonnull;
 import io.quarkus.security.identity.SecurityIdentity;
+import org.eclipse.microprofile.context.ThreadContext;
 import reactor.core.publisher.Flux;
+import reactor.core.scheduler.Schedulers;
 
 @ApplicationScoped
 public class ClockService {
 
     private final SecurityIdentity securityIdentity;
 
-    public ClockService(SecurityIdentity securityIdentity) {
+    public ClockService(SecurityIdentity securityIdentity, ThreadContext threadContext) {
         this.securityIdentity = securityIdentity;
+        Schedulers.onScheduleHook("managed-thread", threadContext::contextualRunnable);
     }
 
     private final String id = UUID.randomUUID().toString();

